argopy/fetchers.py

```diff
@@ -264,29 +312,161 @@ def to_xarray(self, **kwargs):
+            raise InvalidFetcher(
+                " Initialize an access point (%s) first."
+                % ",".join(self.Fetchers.keys())
+            )
         xds = self.fetcher.to_xarray(**kwargs)
         xds = self.postproccessor(xds)
         return xds
 
     def to_dataframe(self, **kwargs):
-        """  Fetch and return data as pandas.Dataframe """
+        """  Fetch and return data as pandas.Dataframe
+
+            Returns
+            -------
+            :class:`pandas.Dataframe`
+        """
         if not self.fetcher:
-            raise InvalidFetcher(" Initialize an access point (%s) first." %
-                                 ",".join(self.Fetchers.keys()))
-        return self.to_xarray(**kwargs).to_dataframe()
+            raise InvalidFetcher(
+                " Initialize an access point (%s) first."
+                % ",".join(self.Fetchers.keys())
+            )
+        return self.load().data.to_dataframe(**kwargs)
+
+    def to_index(self, full: bool = False):
+        """ Create an index of Argo data
+
+            Parameters
+            ----------
+            full: bool
+                Should extract a full index, as returned by an IndexFetcher or only a space/time
+                index of fetched profiles (this is the default choice, i.e. full=False).
+
+            Returns
+            -------
+            :class:`pandas.Dataframe`
+        """
+        if not full:
+            self.load()
+            ds = self.data.argo.point2profile()
+            df = (
+                ds.drop_vars(set(ds.data_vars) - set(["PLATFORM_NUMBER"]))
+                .drop_dims("N_LEVELS")
+                .to_dataframe()
+            )
+            df = (
+                df.reset_index()
+                .rename(
+                    columns={
+                        "PLATFORM_NUMBER": "wmo",
+                        "LONGITUDE": "longitude",
+                        "LATITUDE": "latitude",
+                        "TIME": "date",
+                    }
+                )
+                .drop(columns="N_PROF")
+            )
+            df = df[["date", "latitude", "longitude", "wmo"]]
+
+        else:
+            # Instantiate and load an IndexFetcher:
+            index_loader = ArgoIndexFetcher(mode=self._mode,
+                                            src=self._src,
+                                            ds=self._dataset_id,
+                                            **self.fetcher_kwargs)
+            if self._AccessPoint == 'float':
+                index_loader.float(self._AccessPoint_data['wmo']).load()
+            if self._AccessPoint == 'profile':
+                index_loader.profile(self._AccessPoint_data['wmo'], self._AccessPoint_data['cyc']).load()
+            if self._AccessPoint == 'region':
+                # Convert data box to index box (remove depth info):
+                index_box = self._AccessPoint_data['box']
+                del index_box[4:6]
```

Comment:
`obj.to_index(full=True)` not working on my side. I think that `del index_box[4:6]` also delete elements inside `self._AccessPoint_data['box']` so our DataFetcher proto isn't valid after that, the error occurs in the `self.__repr__()` line 397. Do you have the same behavior ?

Reply:
good spot !
Fixed with: 
```
index_box = self._AccessPoint_data['box'].copy()
```